en/sql/tuning.rst

```diff
@@ -368,6 +385,228 @@ If you ensure that left table's row number is a lot smaller than the right table
     SELECT /*+ RECOMPILE ORDERED */  DISTINCT h.host_year, o.host_nation
     FROM history h INNER JOIN olympic o ON h.host_year = o.host_year;
 
+You can specify the table order by specifying the **LEADING** hint. Unlike the **ORDERED** hint, it can be specified regardless of the order in which the tables are written.
+
+.. code-block:: sql
+
+    SELECT /*+ RECOMPILE LEADING(o,h) */  DISTINCT h.host_year, o.host_nation
+    FROM history h INNER JOIN olympic o ON h.host_year = o.host_year;
+
+.. _optimizer-principle:
+
+Optimizer Principle
+=====================
+
+**CUBRID**\'s optimizer performs cost-based optimization when generating a query execution plan, and calculates **selectivity**\ and **expected number of rows and pages**\ through statistics. Based on this, it calculates the cost and selects the execution plan with the lowest cost.
+
+Selectivity
+--------------
+
+**Selectivity**\ is the ratio of data to be selected when a specific predicate is evaluated. The optimizer calculates the selectivity for each predicate in the **WHERE** clause. **CUBRID**\ assumes that the entire data is evenly distributed and calculates the **selectivity**\ using the **Number of Distinct Value**\ of statistics.
+
+::
+
+    SELECTIVITY = 1 / Number of Distinct Value
+
+.. code-block:: sql
+
+    CREATE TABLE t1 (code INT, name VARCHAR(20));
+    INSERT INTO t1 VALUES(1,'Park'),(2,'Park'),(3,'Park'),(4,'joo'),(5,'joo'),(5,'joo');
+    CREATE INDEX i_t1_code ON t1(code,name);
+    UPDATE STATISTICS ON t1;
+     
+    ;info stats t1
+
+::
+
+    CLASS STATISTICS
+    ****************
+     Class name: t1 Timestamp: Fri Dec 20 13:07:58 2024
+     Total pages in class heap: 1
+     Total objects: 6
+     Number of attributes: 2
+     Attribute: code (integer)
+        Number of Distinct Values: 5
+        B+tree statistics:
+            BTID: { 0 , 5760 }
+            Cardinality: 5 (5,5) , Total pages: 3 , Leaf pages: 1 , Height: 2
+     
+     Attribute: name (character varying)
+        Number of Distinct Values: 2
+
+.. code-block:: sql
+
+    ;plan detail
+    SELECT /*+ recompile */ * FROM t1 WHERE code = 3 AND name IN ('Song', 'Ham');
+
+::
+
+    Join graph terms:
+    term[0]: [dba.t1].code=3 (sel 0.2)
+    term[1]: [dba.t1].[name] range ('Ham' =  or 'Song' = ) (sel 0.75)
+
+**term[0]: [dba.t1].code=3**\ The **Number of Distinct Values**\ of **code**\ in the statistics is 5, so the selectivity is 1/5 or 0.2.
+**term[1]: [dba.t1].[name] range ('Ham' = or 'Song' = )** performs an **OR** operation on two values of name. Since the **Number of Distinct Values** for **name** is 2, the selectivity for each value is 1/2. Adding the two values and subtracting the intersection gives the calculation **0.5 + 0.5 - (0.5 x 0.5)**, resulting in 0.75.
```

Comment:
```suggestion
**term[0]: [dba.t1].code=3** The **Number of Distinct Values** of **code** in the statistics is 5, so the selectivity is 1/5 or 0.2.
**term[1]: [dba.t1].[name] range ('Ham' = or 'Song' = )** performs an **OR** operation on two values of name. Since the **Number of Distinct Values** for **name** is 2, the selectivity for each value is 1/2. Adding the two values and subtracting the intersection gives the calculation **0.5 + 0.5 - (0.5 x 0.5)**, resulting in 0.75.
```